src/libs/routes/errorHandler.ts

```diff
@@ -0,0 +1,12 @@
+import { Request, Response, NextFunction } from 'express';
+import { IError } from './interface';
+export default(err: IError, req: Request, res: Response, next: NextFunction) => {
+    res.json(
```

Comment:
Add status code to headers using res.status(). So it will return proper HTTP code as well.
Before sending the response, check if the response header already sent. If already sent delegate the error to express using next().

For reference visit [Before sending the response, check if the response header already sent. If already sent delegate the error to express using next().

For reference visit https://expressjs.com/en/guide/error-handling.html](https://expressjs.com/en/guide/error-handling.html)

Reply:
ok sir